### PR TITLE
Publish payg-credit and link handshake (IOT-192)

### PR DESCRIPTION
**Pull request by jjmilburn**
These resources are implemented in https://github.com/angaza/nexus-embedded/tree/master/nexus , and demonstrated here: https://www.youtube.com/watch?v=A_SJdJS7SMQ

Introduce the OpenAPI resource models into the registry.

### CODE REVIEW

File: resource_type_registry.md [hunk 1]
```diff
@@ -6,4 +6,6 @@
 | 201   | angaza.com.nexus.core.drlc | [drlc.oas.yaml](resource_types/core/energy/201-drlc/redoc_wrapper.md) |
 | 202   | angaza.com.nexus.core.operatingmode | [operatingmode.oas.yaml](resource_types/core/202-operatingmode/redoc_wrapper.md) |
 | 301   | angaza.com.nexus.core.samplelog | [samplelog.oas.yaml](resource_types/core/301-samplelog/redoc_wrapper.md) |
+| 401   | angaza.com.nexus.core.paygcredit | [paygcredit.oas.yaml](resource_types/core/401-paygcredit/redoc_wrapper.md) |
+| 1001  | angaza.com.nexus.link.handshake | [linkhandshake.oas.yaml](resource_types/link/1001-linkhandshake/redoc_wrapper.md) |
```
Comment (ricehornet) [inline resource_type_registry.md]: FYI we are using different RTRs for these resources in the embedded dev repo.

File: resource_types/core/401-paygcredit/paygcredit.oas.yaml [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        # - $ref: '#/parameters/interface'
+      responses:
+        200:
+          description: "PAYG Credit specific information about this device."
+          schema:
+            $ref: '#/definitions/FullPAYGState'
+          x-example:
+            description: "x-example included only for automatic code generation."
+            rt:
+              - angaza.com.nexus.channel.core.paygcredit
```
Comment (ricehornet) [inline resource_types/core/401-paygcredit/paygcredit.oas.yaml]:
```suggestion
              - angaza.com.nx.pc
```

This is what it is in the embedded dev repo.

File: resource_types/core/401-paygcredit/paygcredit.oas.yaml [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+        400:
+          description: Credit amount/value not supported.
+        401:
+          description: Client is not authorized to perform this action (client may lack a Nexus Channel Link to this device)
+        405:
+          description: This device is not currently capable of remotely updating credit (may be in 'independent' mode)
+          schema:
+              properties:
+                mo:
+                  $ref: '#/definitions/FullPAYGState/properties/mo'
```
Comment (ricehornet) [inline resource_types/core/401-paygcredit/paygcredit.oas.yaml]: I think it does not now actually do this (send `mo` with the error response)?

File: resource_types/link/1001-linkhandshake/linkhandshake.oas.yaml [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+        Retrieve information about a Nexus Channel Link handshake in progress.
+        Only one link handshake may be in progress at any given time, this
+        endpoint provides the information about that handshake.
+      responses:
+        200:
+          description: > 
+            Current Channel Link handshake status
+          x-example:
+            rt: 
+              - angaza.com.nexus.link.handshake
```
Comment (ricehornet) [inline resource_types/link/1001-linkhandshake/linkhandshake.oas.yaml]:
```suggestion
              - angaza.com.nx.ln.hs
```

File: resource_types/link/1001-linkhandshake/linkhandshake.oas.yaml [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+        description: >
+          "Response Data". Binary response data sent to 'challenger' during handshake establishment.
+        type: string
+        format: binary
+        maxLength: 16
+        example: "\x63\xfb\x91\x00\x38"
+      cM:
+        description: >
+          "Challenge Mode".
+          Nexus Channel Link Handshake challenge authentication types supported. These
```
Comment (ricehornet) [inline resource_types/link/1001-linkhandshake/linkhandshake.oas.yaml]:
```suggestion
          Nexus Channel Link Handshake challenge authentication type for this handshake. These
```

File: resource_types/link/1001-linkhandshake/linkhandshake.oas.yaml [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+        format: uint8
+        enum:
+          - 0
+          - 1
+          - 2
+        example: 0
+      lS:
+        description: >
+          "Supported Link Security Modes".
+          Nexus Channel Link security modes supported by this device. These determine
```
Comment (ricehornet) [inline resource_types/link/1001-linkhandshake/linkhandshake.oas.yaml]:
```suggestion
          Nexus Channel Link security mode for this link. These determine
```

File: resource_types/link/1001-linkhandshake/linkhandshake.oas.yaml [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+  LinkHandshake:
+    type: object
+    properties:
+      rt:
+        description: |
+          **Fixed Specification Value Only**. The Resource Type
+          (Only reported if `oic.if.baseline` interface query is supported)
+        items:
+          enum:
+          - angaza.com.nexus.link.handshake
```
Comment (ricehornet) [inline resource_types/link/1001-linkhandshake/linkhandshake.oas.yaml]:
```suggestion
          - angaza.com.nx.ln.hs
```

**Review by ricehornet — APPROVED** [review]
@jjmilburn LGTM, just have a few minor requests to improve clarity and sync with the embedded repo.

File: resource_type_registry.md (hunk 1 above)
Comment (jjmilburn) [reply resource_type_registry.md]: Updating on embedded side.

Comment (jjmilburn) [reply resource_type_registry.md]: see - https://github.com/angaza/nexus-keycode-embedded-internal/pull/140

File: resource_types/core/401-paygcredit/paygcredit.oas.yaml (hunk 3 above)
Comment (jjmilburn) [reply resource_types/core/401-paygcredit/paygcredit.oas.yaml]: Will delete, to add as a later addition if needed.

Comment (jjmilburn) [reply resource_types/core/401-paygcredit/paygcredit.oas.yaml]: Updated to remove this

**Pull request merged**